tests/utils.py

```diff
@@ -196,19 +196,21 @@ def from_call_to_call_array(calls):
         calldata.extend(call[2])
     return (call_array, calldata)
 
-
-def hash_multicall(sender, calls, nonce, max_fee):
-    hash_array = []
-    for call in calls:
-        call_elements = [call[0], call[1], compute_hash_on_elements(call[2])]
-        hash_array.append(compute_hash_on_elements(call_elements))
-
-    message = [
-        str_to_felt('StarkNet Transaction'),
-        sender,
-        compute_hash_on_elements(hash_array),
-        nonce,
+def get_transaction_hash(account, call_array, calldata, nonce, max_fee):
+    execute_calldata = [
+        len(call_array),
+        *[x for t in call_array for x in t],
+        len(calldata),
+        *calldata,
+        nonce]
+
+    return calculate_transaction_hash_common(
+        TransactionHashPrefix.INVOKE,
+        TRANSACTION_VERSION,
+        account,
+        get_selector_from_name('__execute__'),
+        execute_calldata,
         max_fee,
-        TRANSACTION_VERSION
-    ]
-    return compute_hash_on_elements(message)
+        StarknetChainId.TESTNET.value,
```

Comment:
where does this value come from? will this always be `TESTNET` or when should we change it?

Reply:
There is only 2 values as defined in https://github.com/starkware-libs/cairo-lang/blob/4e233516f52477ad158bc81a86ec2760471c1b65/src/starkware/starknet/definitions/general_config.py#L31 . The testing `starknet` object is using `TESNET`.

The signing client (e.g. Argent X) will need to use the correct value depending on the chain.